File: src/main/java/com/lakshan/medi_sync/entity/LipidProfile.java
package com.lakshan.medi_sync.entity;

import jakarta.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "lipid_profile")
public class LipidProfile {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private int id;

    @Column(name = "test_date")
    private LocalDate testDate;

    @Column(name = "total_cholesterol")
    private double totalCholesterol;

    @Column(name = "hdl")
    private double hdl;

    @Column(name = "ldl")
    private double ldl;

    @Column(name = "vldl")
    private double vldl;

    @Column(name = "triglycerides")
    private double triglycerides;

    @Column(name = "image_url")
    private String imageUrl;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public LocalDate getTestDate() {
        return testDate;
    }

    public void setTestDate(LocalDate testDate) {
        this.testDate = testDate;
    }

    public double getTotalCholesterol() {
        return totalCholesterol;
    }

    public void setTotalCholesterol(double totalCholesterol) {
        this.totalCholesterol = totalCholesterol;
    }

    public double getHdl() {
        return hdl;
    }

    public void setHdl(double hdl) {
        this.hdl = hdl;
    }

    public double getLdl() {
        return ldl;
    }

    public void setLdl(double ldl) {
        this.ldl = ldl;
    }

    public double getVldl() {
        return vldl;
    }

    public void setVldl(double vldl) {
        this.vldl = vldl;
    }

    public double getTriglycerides() {
        return triglycerides;
    }

    public void setTriglycerides(double triglycerides) {
        this.triglycerides = triglycerides;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }
}

File: lib/models/lipid_profile.dart
import 'user.dart';

class LipidProfile {
  final int id;
  final String testDate;
  final double totalCholesterol;
  final double hdl;
  final double ldl;
  final double triglycerides;
  final String? imageUrl;
  final User user;

  LipidProfile({
    required this.id,
    required this.testDate,
    required this.totalCholesterol,
    required this.hdl,
    required this.ldl,
    required this.triglycerides,
    this.imageUrl,
    required this.user,
  });

  factory LipidProfile.fromJson(Map<String, dynamic> json) {
    return LipidProfile(
      id: json['id'] as int,
      testDate: json['testDate'] as String,
      totalCholesterol: (json['totalCholesterol'] as num).toDouble(),
      hdl: (json['hdl'] as num).toDouble(),
      ldl: (json['ldl'] as num).toDouble(),
      triglycerides: (json['triglycerides'] as num).toDouble(),
      imageUrl: json['imageUrl'] as String?,
      user: User.fromJson(json['user'] as Map<String, dynamic>),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'testDate': testDate,
      'totalCholesterol': totalCholesterol,
      'hdl': hdl,
      'ldl': ldl,
      'triglycerides': triglycerides,
      'imageUrl': imageUrl,
      'user': user.toJson(),
    };
  }

  Map<String, dynamic> toCreateJson(int userId) {
    return {
      'testDate': testDate,
      'totalCholesterol': totalCholesterol,
      'hdl': hdl,
      'ldl': ldl,
      'triglycerides': triglycerides,
      'imageUrl': imageUrl,
      'userId': userId,
    };
  }
}
